x/mongo/driver/topology/connection.go

```diff
@@ -717,21 +724,27 @@ func (c *Connection) LocalAddress() address.Address {
 
 // PinToCursor updates this connection to reflect that it is pinned to a cursor.
 func (c *Connection) PinToCursor() error {
-	return c.pin("cursor")
+	return c.pin("cursor", c.pool.pinConnectionToCursor, c.pool.unpinConnectionFromCursor)
 }
 
 // PinToTransaction updates this connection to reflect that it is pinned to a transaction.
 func (c *Connection) PinToTransaction() error {
-	return c.pin("transaction")
+	return c.pin("transaction", c.pool.pinConnectionToTransaction, c.pool.unpinConnectionFromTransaction)
 }
 
-func (c *Connection) pin(reason string) error {
+func (c *Connection) pin(reason string, updatePoolFn, cleanupPoolFn func()) error {
```

Comment:
Cool way of passing cursor and transaction logic... To be clear, these will only ever be `(un)pinConnectionToCursor` or `(un)pinConnectionToTransaction`, right?

Reply:
Yep.